### PR TITLE
LA-10679 Fix installer script to use latest kubernetes package repo

### PR DESCRIPTION
**Pull request by tushar-lb**
https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/#how-to-migrate

### CODE REVIEW

File: Ubuntu/master.sh [hunk 1]
```diff
@@ -139,10 +126,9 @@ function serviceStatusCheck() {
 echo "2. Install kubeadm, kubectl and kubelet:"
 apt-get update -y && apt-get install -y openssh-server apt-transport-https curl
 curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
-deb http://apt.kubernetes.io/ kubernetes-xenial main
-EOF
-apt-get update -y && apt-get install -y kubelet=1.25.0-00 kubeadm=1.25.0-00 kubectl=1.25.0-00
+mkdir -p /etc/apt/keyrings
+echo "deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.28/deb/ /" | sudo tee /etc/apt/sources.list.d/kubernetes.list
+apt-get update -y && apt-get install -y kubelet=1.28.8-1.1 kubeadm=1.28.8-1.1 kubectl=1.28.8-1.1
```
Comment (ajay-lb) [inline Ubuntu/master.sh:131]: Will QA also move to test with 1.28?

File: Ubuntu/worker.sh [hunk 2]
```diff
@@ -137,9 +124,23 @@ function serviceStatusCheck() {
 echo "2. Install kubeadm, kubectl and kubelet:"
 apt-get update -y && apt-get install -y openssh-server apt-transport-https curl
```
Comment (ajay-lb) [inline Ubuntu/worker.sh:125]: Should this just be merged with line 16?

File: Ubuntu/master.sh [hunk 3]
```diff
@@ -139,10 +126,9 @@ function serviceStatusCheck() {
 echo "2. Install kubeadm, kubectl and kubelet:"
 apt-get update -y && apt-get install -y openssh-server apt-transport-https curl
```
Comment (ajay-lb) [inline Ubuntu/master.sh:127]: should this be merged into line 16?

**Review by ajay-lb — APPROVED** [review]

File: Ubuntu/master.sh [hunk 4]
```diff
@@ -5,26 +5,13 @@ if [ "$EUID" -ne 0 ]; then
...
-apt-mark hold docker-compose-plugin
-apt-mark hold snapd
-apt-mark hold systemd
-apt-mark hold systemd-sysv
-apt-mark hold systemd-timesyncd
-
 TIMEOUT=300
 SLEEP_INTERVAL=1
 
+# Remove the older packages.
```
Comment (aditya-lb) [inline Ubuntu/master.sh:11]:
```suggestion
# Remove all older packages.
```

We should be a bit more intelligent and remove them only if they are not the expected version. We should ensure that this script is idempotent where if we run it multiple times, we don't do wasted work.

File: Ubuntu/master.sh [hunk 5]
```diff
@@ -183,3 +169,19 @@ apt install python3-pip -y
 # set default namespace as lightbeam
 kubectl config set-context --current --namespace lightbeam
 echo "Done! Ready to deploy LightBeam Cluster!!"
+
+
+# Mark packages on hold to avoid auto upgrade.
```
Comment (aditya-lb) [inline Ubuntu/master.sh:174]:
```suggestion
# Mark packages on hold to avoid an auto upgrade.
```

File: Ubuntu/worker.sh [hunk 6]
```diff
@@ -5,26 +5,13 @@ if [ "$EUID" -ne 0 ]; then
...
-apt-mark hold snapd
-apt-mark hold systemd
-apt-mark hold systemd-sysv
-apt-mark hold systemd-timesyncd
-
 TIMEOUT=300
 SLEEP_INTERVAL=1
 
+# Remove the older packages.
+apt-get -y remove docker docker-engine docker.io containerd runc kubeadm kubelet kubectl
```
Comment (aditya-lb) [inline Ubuntu/worker.sh:12]: Same comment as above.

**Review by aditya-lb — CHANGES_REQUESTED** [review]

File: Ubuntu/master.sh (hunk 4 above)
Comment (tushar-lb) [reply Ubuntu/master.sh]: We can add additional check before remove in separate PR.

**Pull request merged**